numpy/core/tests/test__exceptions.py

```diff
@@ -40,3 +44,9 @@ def test__total_size(self):
 
         e = _ArrayMemoryError((2, 4), np.dtype((np.uint64, 16)))
         assert e._total_size == 1024
+
+
+class TestUFuncNoLoopError:
+    def test_pickling(self):
+        """ Test that _UFuncNoLoopError can be pickled """
+        assert isinstance(pickle.dumps(_UFuncNoLoopError), bytes)
```

Comment:
Thanks for looking into this, looks good to me.  If you have some time, maybe you can extend the test to also loading and the error instance and not just the error class?

It may be easier for memory error (either seems enough to me for now). Could you add a test that round-tripping works, maybe something like:
```
error = _ArrayMemoryError((1023,), np.dtype(np.uint8))
res = pickle.loads(pickle.dumps(error))

assert res._total_size == error._total_size
```
I would mostly like a test that the error (instances) can also be unpickled again.  I admit, they clearly can here.

Reply:
done in 866b39e8217eb1b7d2215a9a462bdfd86f4f516e

Reply:
Thanks. One more nitpick: test would make more sense as a `test_pickling` in the `TestArrayMemoryError` class.

Reply:
fixed in ea0c5f5295650ee99856133565b98b3e99396fb2